mxcubeweb/core/components/samplechanger.py

```diff
@@ -467,66 +509,85 @@ def queue_mount_sample(view, data_model, centring_done_cb, async_result):  # noq
+                        centring_method = queue_entry.CENTRING_METHOD
+
+                        if centring_method == queue_entry.CENTRING_METHOD.MANUAL:
+                            msg = (
+                                "Manual centring used, waiting for"
+                                + " user to center sample"
                             )
+                            log.warning(msg)
+                            dm.start_centring_method(dm.MANUAL3CLICK_MODE)
+                        elif centring_method in [
+                            queue_entry.CENTRING_METHOD.LOOP,
+                            queue_entry.CENTRING_METHOD.FULLY_AUTOMATIC,
+                        ]:
+                            if not dm.current_centring_procedure:
+                                dm.start_centring_method(dm.C3D_MODE)
+
+                            # NBNB  BUG . self and app are not avialble here
+                            if mxcube.AUTO_MOUNT_SAMPLE:
+                                msg = (
+                                    "Going to save centring automatically, please wait"
+                                )
+                            else:
+                                msg = (
+                                    "Centring in progress. Please save"
+                                    + " the suggested centring or re-center"
+                                )
+
+                            log.warning(msg)
                         else:
-                            raise RuntimeError("Could not center sample")
-                except Exception:
-                    import traceback
+                            dm.start_centring_method(dm.MANUAL3CLICK_MODE)
 
-                    log.info("centring did not pass %s" % traceback.format_exc())
-                finally:
-                    dm.disconnect("centringAccepted", centring_done_cb)
-        else:
-            dm.disconnect("centringAccepted", centring_done_cb)
+                        logging.getLogger("user_level_log").info("Centring ...")
+                        centring_result = async_result.get()
+                        if centring_result["valid"]:
+                            logging.getLogger("user_level_log").info("Centring done !")
+                        else:
+                            if (
+                                centring_method
+                                == queue_entry.CENTRING_METHOD.FULLY_AUTOMATIC
+                            ):
+                                raise queue_entry.QueueSkippEntryException(
+                                    "Could not center sample, skipping",
+                                    "",
+                                )
+                            else:
+                                raise RuntimeError("Could not center sample")
+                    except Exception:
+                        import traceback
+
+                        log.info("centring did not pass %s" % traceback.format_exc())
```

Comment:
This is probably better:

```suggestion
                        log.exception("centring did not pass")
```

The exception including traceback should be automatically added: https://docs.python.org/3/library/logging.html#logging.exception